exec/vector/src/main/java/org/apache/drill/exec/util/DecimalUtility.java

```diff
@@ -159,9 +159,20 @@ public static BigDecimal getBigDecimalFromSparse(DrillBuf data, int startIndex,
     }
 
     public static BigDecimal getBigDecimalFromDrillBuf(DrillBuf bytebuf, int start, int length, int scale) {
+      if (length <= 0) {
+        // if the length is somehow non-positive, interpret this as zero
+        //System.out.println("getBigDecimal forces 0 with start " + start + " len " + length);
+          try {
+              throw new Exception("hi there");
```

Comment:
Please remove

Reply:
Yes, I'll remove the friendly "hi there".  It was for debugging, I guess.